app/api/auth/agent-connect/agent-connect-types.ts

```diff
@@ -30,3 +30,27 @@ export class AgentConnectCouldBeAServicePublicException extends Exception {
     });
   }
 }
+
+export class AgentConnectMCPNoSiretException extends Exception {
+  constructor(message: string, userId: string) {
+    super({
+      name: 'AgentConnectMCPNoSiretException',
+      message,
+      context: {
+        details: userId,
+      },
+    });
+  }
+}
+
+export class AgentConnectProviderNoSiretException extends Exception {
+  constructor(message: string, idpId: string) {
+    super({
+      name: 'AgentConnectProviderNoSiretException',
+      message,
+      context: {
+        details: idpId,
+      },
+    });
+  }
+}
```

Comment:
Why not use `AgentConnectFailedException`. Wont the way you currently throw it result in an agent connection failed ?

Reply:
I thought it might be better to have different types of events in Sentry to clearly see what goes wrong.